apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/r4/providers/CoverageTransformerV2Test.java

```diff
@@ -0,0 +1,271 @@
+package gov.cms.bfd.server.war.r4.providers;
+
+import static org.hamcrest.CoreMatchers.*;
+
+import ca.uhn.fhir.context.FhirContext;
+import com.codahale.metrics.MetricRegistry;
+import gov.cms.bfd.model.codebook.data.CcwCodebookVariable;
+import gov.cms.bfd.model.rif.Beneficiary;
+import gov.cms.bfd.model.rif.samples.StaticRifResource;
+import gov.cms.bfd.model.rif.samples.StaticRifResourceGroup;
+import gov.cms.bfd.server.war.ServerTestUtils;
+import gov.cms.bfd.server.war.commons.MedicareSegment;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import org.hl7.fhir.exceptions.FHIRException;
+import org.hl7.fhir.r4.model.Coverage;
+import org.hl7.fhir.r4.model.Coverage.CoverageStatus;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/** Unit tests for {@link gov.cms.bfd.server.war.stu3.providers.CoverageTransformerV2}. */
+public final class CoverageTransformerV2Test {
+
+  private static final FhirContext fhirContext = FhirContext.forR4();
+  private static Beneficiary beneficiary = null;
+
+  @Before
+  public void setup() {
+    List<Object> parsedRecords =
+        ServerTestUtils.parseData(Arrays.asList(StaticRifResourceGroup.SAMPLE_A.getResources()));
+
+    // Pull out the base Beneficiary record and fix its HICN and MBI-HASH fields.
+    beneficiary =
+        parsedRecords.stream()
+            .filter(r -> r instanceof Beneficiary)
+            .map(r -> (Beneficiary) r)
+            .findFirst()
+            .get();
+    beneficiary.setLastUpdated(new Date());
+  }
+
+  @After
+  public void tearDown() {
+    beneficiary = null;
+  }
+
+  /**
+   * Verifies that {@link
+   * gov.cms.bfd.server.war.stu3.providers.CoverageTransformerV2#transform(MedicareSegment,
+   * Beneficiary)} works as expected when run against the {@link StaticRifResource#SAMPLE_A_CARRIER}
+   * {@link Beneficiary}.
+   *
+   * @throws FHIRException (indicates test failure)
+   */
+  @Test
+  public void transformSampleARecord() throws FHIRException {
```

Comment:
This is more personal preference, but I'd break this into five separate tests.  One for each A-D and one for the "null" time.

Reply:
I did break them apart; better that way. thx.